src/aspire/ctf/ctf_estimator.py

```diff
@@ -0,0 +1,757 @@
+"""
+Created on Sep 10, 2019
+@author: Ayelet Heimowitz, Amit Moscovich
+
+Integrated into ASPIRE by Garrett Wright Feb 2021.
+"""
+
+import logging
+import os
+
+import numpy as np
+from numpy import linalg as npla
+from scipy.optimize import linprog
+
+from aspire.image import Image
+from aspire.numeric import fft
+from aspire.utils import complex_type, eigs
+
+logger = logging.getLogger(__name__)
+
+
+def abs2(x):
+    """
+    Compute complex modulus squared.
+    """
+
+    return x.real ** 2 + x.imag ** 2
+
+
+class CtfEstimator:
+    """
+    CtfEstimator Class ...
+    """
+
+    def __init__(
+        self,
+        pixel_size,
+        cs,
+        amplitude_contrast,
+        voltage,
+        psd_size,
+        num_tapers,
+        dtype=np.float32,
+    ):
+        """
+        Instantiate a CtfEstimator instance.
+
+        :param pixel_size: Size of the pixel in Angstrom.
+        :param cs: Spherical aberration in mm.
+        :param amplitude_contrast: Amplitude contrast.
+        :param voltage: Voltage of electron microscope.
+        :param psd_size: Block size (in pixels) for PSD estimation.
+        :param num_tapers: Number of tapers to apply in PSD estimation.
+        :returns: CtfEstimator instance.
+        """
+
+        self.pixel_size = pixel_size
+        self.cs = cs
+        self.amplitude_contrast = amplitude_contrast
+        self.voltage = voltage
+        self.psd_size = psd_size
+        self.num_tapers = num_tapers
+        self.lmbd = 1.22639 / np.sqrt(voltage * 1000 + 0.97845 * np.square(voltage))
+        self.dtype = np.dtype(dtype)
+
+        center = psd_size // 2
```

Comment:
Just changed this.  We can, but there are different conventions...  This code was -half to half, while grid_2d is -1,1.  Here it doesn't matter because we're just going to grab `phi` for thetas.  But, also the mesh is transposed :joy:.  Obviously we can fix this here, but it might crop up less trivially other changes... we'll see.

Reply:
Right. Part of what we should do is to make sure that we use the same grid conventions throughout. Things like [-1, 1] vs. [-½, ½] are not too big of a deal, but things like including endpoints (left or right) and centering for odd/even can be tricky.